src/main/java/fi/metatavu/metamind/bot/slots/LocalizedNumberSlot.java

```diff
@@ -0,0 +1,198 @@
+  private Number localizedWordStringToNumber(String token) {
+    for (String language : SUPPORTED_LANGUAGES) {
+      Number result = null;
+      switch (language) {
+        case "finnish":
+          result = finnishWordStringToNumber(token);
+        break;
+        case "english":
+          result = englishWordStringToNumber(token);
+        break;
+      }
+      
+      if (result != null) {
+        return result;
+      }
+    }
+
+    return null;
+  }
+  
+  private Number englishWordStringToNumber(String token) {
+    return wordStringToNumber(token);
+  }
+  
+  private Number finnishWordStringToNumber(String token) {
+    if (token == null || token.length() < 1)
+    {
+      return null;
+    }
+
+    token = token.replaceAll("-", " ");
+    token = token.replaceAll(",", " ");
+    token = token.toLowerCase().replaceAll(" ja", " ");
+    String[] splittedParts = token.trim().split("\\s+");
+
+    long result = 0;
+
+    for (String str : splittedParts) {
+      str = str.toLowerCase();
+      switch (str) {
+        case "nolla":
+          result += 0;
+        break;
+        case "yksi":
+          result += 1;
+        break;
+        case "kaksi":
+          result += 2;
+        break;
+        case "kolme":
+          result += 3;
+        break;
+        case "neljä":
+          result += 4;
+        break;
+        case "viisi":
+          result += 5;
+        break;
+        case "kuusi":
+          result += 6;
+        break;
+        case "seitsemän":
+          result += 7;
+        break;
+        case "kahdeksan":
+          result += 8;
+        break;
+        case "yhdeksän":
+          result += 9;
+        break;
+        case "kymmenen":
+          result += 10;
+        break; 
+        case "yksitoista":
+          result += 11;
+        break; 
+        case "kaksitoista":
+          result += 12;
+        break;
+        case "kolmetoista":
+          result += 13;
+        break;
+        case "neljätoista":
+          result += 14;
+        break;
+        case "viisitoista":
+          result += 15;
+        break;
+        case "kuusitoista":
+          result += 16;
+        break;
+        case "seitsemäntoista":
+          result += 17;
+        break;
+        case "kahdeksantoista":
+          result += 18;
+        break;
+        case "yhdeksäntoista":
+          result += 19;
+        break;
+        case "kaksikymmentä":
+          result += 20;
+        break;
+        case "kolmekymmentä":
+          result += 30;
+        break;
+        case "neljäkymmentä":
+          result += 40;
+        break;
+        case "viisikymmentä":
+          result += 50;
+        break;
+        case "kuusikymmentä":
+          result += 60;
+        break;
+        case "seitsemänkymmentä":
+          result += 70;
+        break;
+        case "kahdeksankymmentä":
+          result += 80;
+        break;
+        case "yhdeksänkymmentä":
+          result += 90;
+        break;
+        case "sata":
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') End this switch case with an unconditional break, return or throw statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS128)